### PR TITLE
[C-4853] - support v2 syntax nodejs sdk

### PR DESCRIPTION
**Pull request by cgradwohl**
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/send/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    if((params as ICourierSendV2Parameters).message) {
+      const v2Response = await sendV2Call(options, axiosConfig, (params as ICourierSendV2Parameters));
+      return v2Response as SendResponse<T>;
+    }
+
+    const v1Response = await sendCall(options, axiosConfig, (params as ICourierSendParameters));
+    return v1Response as SendResponse<T>;
+  };
+};
```
Comment (drew-y) [inline src/send/index.ts:60]: Nitpick. I recommend configuring VSCode add a newline at the end or your files on save. Makes it so `cat` prints neatly without screwing with the prompt.

File: src/send/types.ts [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+  elements: ElementalNode[];
+}
+
+export interface ElementalQuoteNode extends ElementalBaseNode {
+  type: "quote";
+  content: string;
+  align?: IAlignment;
+  borderColor?: string;
+  textStyle?: "text" | "h1" | "h2" | "subtext";
+  locales?: ElementalLocales;
```
Comment (drew-y) [inline src/send/types.ts:172]: We should avoid exporting the ElementalLocales. It's not exactly spec and right now it is specific to elements with a content property. I'm worried we're asking for a breaking change in the future here.

```suggestion
  locales?: {
    [locale: string]: {
      content: string;
    };
  };
```

File: src/send/types.ts [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+
+export interface ElementalActionNode extends ElementalBaseNode {
+  type: "action";
+  content: string;
+  href: string;
+  actionId?: string;
+  style?: IActionButtonStyle;
+  align?: IAlignment;
+  backgroundColor?: string;
+  locales?: ElementalLocales;
```
Comment (drew-y) [inline src/send/types.ts:153]: See comment below.

```suggestion
  locales?: {
    [locale: string]: {
      content: string;
    };
  };
```

File: src/send/types.ts [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+  | ElementalActionNode
+  | ElementalDividerNode
+  | ElementalGroupNode
+  | ElementalQuoteNode;
+
+export interface ElementalTextNode extends ElementalBaseNode {
+  type: "text";
+  content: string;
+  format?: "markdown";
+  locales?: ElementalLocales;
```
Comment (drew-y) [inline src/send/types.ts:121]: See comment below (on the quote node)

```suggestion
  locales?: {
    [locale: string]: {
      content: string;
    };
  };
```

File: src/send/types.ts [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+  ref?: string;
+  if?: string;
+  loop?: string;
+}
+
+export interface ElementalLocales {
+  [locale: string]: {
+    content: string;
+  };
+}
```
Comment (drew-y) [inline src/send/types.ts:187]: Not in the spec. Might break 😅
```suggestion
```

File: src/send/index.ts (hunk 1 above)
Comment (tk26) [inline src/send/index.ts:60]: v2 API is the internal nomenclature, for the outside world its not really a send v2 so I think we should avoid exposing Node.js SDK types with the word v2 in it (like sendV2call , ICourierSendV2Response , ICourierSendV2Parameters and so on) coz its public facing. SDK should abstract v2 into a regular /send just like the API does

**Review by tk26 — CHANGES_REQUESTED** [review]

**Comment by tk26** [comment]
Lets also make sure we've run `yarn lint && yarn test` and `yarn format` [check Contributing.md] - and we should bump the release version as well.

File: src/send/index.ts (hunk 1 above)
Comment (cgradwohl) [reply src/send/index.ts:60]: sendCall and sendV2Call are private functions

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@trycourier/courier",
-  "version": "3.2.1",
+  "version": "3.3.0",
```
Comment (tk26) [inline package.json:3]: @cgradwohl mind rebasing and changing this to 3.4.0? I used 3.3.0 in bulk processing sorrey

File: src/send/index.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      profile: params.profile,
+      recipient: params.recipientId
+    },
+    config
+  );
+
+  return res.data;
+};
+
+const sendV2Call = async (
```
Comment (tk26) [inline src/send/index.ts:34]: can we rename to not have v2 in it?

File: package.json (hunk 6 above)
Comment (tk26) [reply package.json:3]: also can we update CHANGELOG?

Comment (cgradwohl) [reply package.json:3]: yeah sure

File: src/send/index.ts (hunk 7 above)
Comment (cgradwohl) [reply src/send/index.ts:34]: yeah sure but this is provate

**Review by tk26 — APPROVED** [review]

**Pull request merged**